### PR TITLE
Add --report=json option and more tests

### PR DESCRIPTION
**Pull request by kachkaev**
Closes https://github.com/jfairbank/run-elm/issues/10

### CODE REVIEW

**Comment by kachkaev** [comment]
This PR should be now ready for your review @jfairbank! I improved a few other bits given the opportunity, please see the git log for details.

Unfortunately, Travis CI failed initially because of one line in `.travis.yml`, the purpose of which I did not quite understand:

https://github.com/jfairbank/run-elm/blob/dbf57dd3d3d79e023d746fb7558f5843cc461a7c/.travis.yml#L26

As a result some surplus output was added to `stderr` and [broke some assertions](https://travis-ci.org/jfairbank/run-elm/builds/354958421). Removing `echo` from `.travis.yml` in https://github.com/jfairbank/run-elm/pull/12/commits/48fd2278a6a72574463309d45f50976c491419d2 fixed this bug.

I also removed testing in node 7 and added not 9 since node 7 support has ended. See https://github.com/nodejs/Release

**Comment by kachkaev** [comment]
ping @jfairbank 😃 

WDYT?

File: package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "run-elm": "lib/index.js"
   },
   "scripts": {
+    "clean-test-case-projects": "cd test/integration && git clean -d -f -X",
```
Comment (jfairbank) [inline package.json]: Can we call this `test:clean` instead?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
+## 2.2.0
+
+### NEW - Option `--report`
+
+Passing `--report=json` to run-elm makes Elm compilation errors `stderr` easier to parse ([#12](https://github.com/jfairbank/run-elm/pull/12))
```
Comment (jfairbank) [inline CHANGELOG.md]: > ...Elm compilation errors `stderr` easier to parse...

I think there is a word missing between `errors` and `` `stderr` ``.

**Review by jfairbank — CHANGES_REQUESTED** [review]
Looks good! I just had a couple of minor things.

**Comment by kachkaev** [comment]
Many thanks for your review @jfairbank! Both issues are addressed now 🕺

**Review by jfairbank — APPROVED** [review]

**Comment by jfairbank** [comment]
Thanks again! I released v2.2.0 with these changes.

**Pull request merged**